crates/services/executor/src/executor.rs

```diff
@@ -549,14 +566,26 @@ where
             Ok(())
         };
 
-        while iter.peek().is_some() {
-            for transaction in iter {
-                execute_transaction(&mut *execution_data, transaction)?;
+        let relayed_tx_iter = forced_transactions.into_iter();
+        for transaction in relayed_tx_iter {
+            const RELAYED_GAS_PRICE: Word = 0;
+            execute_transaction(&mut *execution_data, transaction, RELAYED_GAS_PRICE)?;
+        }
+
+        let remaining_gas_limit = block_gas_limit.saturating_sub(execution_data.used_gas);
+
+        // L2 originated transactions should be in the `TxSource`. This will be triggered after
+        // all relayed transactions are processed.
+        let mut regular_tx_iter = source.next(remaining_gas_limit).into_iter().peekable();
+        while regular_tx_iter.peek().is_some() {
+            for transaction in regular_tx_iter {
+                execute_transaction(&mut *execution_data, transaction, gas_price)?;
```

Comment:
And here we can handle skipped transactions, in this case `execute_transaction` will be cleaner=)

<img width="651" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/356a9e0f-2896-433a-8bd7-2c7e8d3e60df">